central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -348,7 +373,19 @@ func (s *serviceImpl) GetComplianceScanCheckResult(ctx context.Context, req *v2.
 		return nil, errors.Wrapf(errox.NotFound, "compliance check result with id %q does not exist", req.GetId())
 	}
 
-	return storagetov2.ComplianceV2CheckResult(scanResult), nil
+	// Check the Compliance Scan object to get the scan time.
+	scanQuery := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanRef, scanResult.GetScanRefId()).
+		ProtoQuery()
+	scans, err := s.scanDS.SearchScans(ctx, scanQuery)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to retrieve scan data for result %q", req.GetId())
+	}
+	// There should only be a single object for a scan ref id
+	if len(scans) != 1 {
+		return nil, errors.Errorf("Unable to retrieve scan data for result %q", req.GetId())
```

Comment:
Better to return this error as errors.Wrapf(errox.NotFound,"Unable to retrieve scan data for result %q", req.GetId())

Reply:
It could be that we have too many and some other data integrity issue.  I'd rather not split hairs on the difference between finding not enough vs finding too many simply for an error message.